robot-server/robot_server/service/legacy/routers/settings.py

```diff
@@ -193,11 +194,20 @@ async def get_settings_reset_options() -> FactoryResetOptions:
     "/settings/reset", description="Perform a factory reset of some robot data"
 )
 async def post_settings_reset_options(
-    factory_reset_commands: Dict[reset_util.ResetOptionId, bool]
+    factory_reset_commands: Dict[reset_util.ResetOptionId, bool],
+    persistence_resetter: PersistenceResetter = Depends(get_persistence_resetter),
 ) -> V1BasicResponse:
-    options = set(k for k, v in factory_reset_commands.items() if v)
+    options = set(
+        k
+        for k, v in factory_reset_commands.items()
+        if v and k != reset_util.ResetOptionId.runs_history
+    )
     reset_util.reset(options)
 
+    if ("runsHistory", True) in factory_reset_commands.items():
```

Comment:
Should this be a `dict.get`, instead? I'd like to avoid the magic strings and loop over all items

```suggestion
    if factory_reset_commands.get(reset_util.ResetOptionId.runs_history, False):
```